src/ensembl/src/shared/components/view-in-app/ViewInApp.test.tsx

```diff
@@ -62,25 +70,32 @@ describe('<ViewInApp />', () => {
   });
 
   it('returns null if the links prop is empty', () => {
-    const wrapper = renderComponent({ links: {} });
-    expect(wrapper.html()).toBe(null);
+    const { container } = renderComponent({ links: {} });
+    expect(container.innerHTML).toBeFalsy();
   });
 
   it('renders correct number of buttons', () => {
-    const wrapper = renderComponent({ links });
-    expect(wrapper.find(ImageButton)).toHaveLength(Object.keys(links).length);
+    const { container } = renderComponent({ links });
+    expect(container.querySelectorAll('button')).toHaveLength(
+      Object.keys(links).length
+    );
   });
 
-  it('switches to correct url when clicked', () => {
-    const wrapper = renderComponent({ links });
+  it('calls replace with the passed links', () => {
+    const replaceFn = jest.spyOn(router, 'replace');
+
+    renderComponent({ links, shouldReplaceState: true });
+
     tuplesSample.forEach(([appName, link]) => {
-      const imageButton = wrapper
-        .find(AppButton)
-        .findWhere((element) => element.prop('appId') === appName)
-        .find(ImageButton);
+      const appButtonContainer = screen.getByTestId(appName);
 
-      imageButton.simulate('click');
-      expect(push).toHaveBeenCalledWith(link);
+      userEvent.click(
+        appButtonContainer.querySelector('button') as HTMLButtonElement
+      );
+
+      expect(replaceFn).toHaveBeenCalledWith(link);
     });
+
+    replaceFn.mockRestore();
   });
```

Comment:
@azangru @msmanoj This test case unfortunately fails with an error: `Actions must be plain objects. Use custom middleware for async actions.` Appreciate any suggestions on how it can be fixed :)

Reply:
You've mocked the push and the replace functions from connected react router with just `jest.fn()`. Which means that these functions will now return `undefined`. Naturally, redux doesn't like it. It wants action creators to return actions that it can handle (normally — objects with the `type` field).

Reply:
@azangru Thanks I'll try it that way

Reply:
I've got the test to work even though typescript doesn't like the signature of the function I've used to mock `replace`

Reply:
Pushed a commit to slightly update the test. I cheated, and used the `any` type 😄  But we don't really care whether the mock `replace` function actually returns the realistic replace action; nothing in our test depends on that. As long as it returns an object, we should be fine 😄 

I also updated the `toHaveBeenCalledWith` to `toHaveBeenLastCalledWith`.

Reply:
@azangru Thanks. It looks much better now with less code 😄

Reply:
Now that we have a test for `replace`, I couldn't resist adding a test for `push` (and simplified these tests slightly, removing the uncertainty in the number of rendered buttons — that is already checked in the previous tests).

Reply:
Awesome. It looks much better now 😃